src/kv/committable_tx.h

```diff
@@ -44,14 +46,27 @@ namespace kv
       }
 
       auto e = store->get_encryptor();
-      auto entry_type = claims_digest.empty() ? EntryType::WriteSet :
-                                                EntryType::WriteSetWithClaims;
+      auto commit_nonce = e->get_commit_nonce({commit_view, version});
```

Comment:
As discussed, this now required the encryptor to be set on the store, even when serialising public maps only (which has been useful for some unit tests). Should we only set this if the encryptor exists, which would make the commit evidence optional, which may complicate some other tests/checks.

Reply:
That's a good question, and I don't know. I've been adding encryptors everywhere for now, because it seems easier than making the tests subtle, and less likely to go wrong, but I'm open to ideas.

Reply:
Yeah, we might as well bite the bullet now and make the encryptor mandatory on the store OR we can set a default commit once (i.e. `00000...00`) if the encryptor isn't set?

Reply:
I've gone with setting the null encryptor where necessary for two reasons as discussed:

1. I don't think it's a very good default for a normal user.
2. It's easier to catch the exception where it's not set than deal later on with unexpectedly incorrect digests.